### DISCUSSION TITLE
command to "merge" workspaces

### DISCUSSION THREAD

**Discussion by SomeGuyNamedMay** [YaLTeR/niri#50] [Ideas]
whats your thoughts on adding functionallity to combine workspaces of windows, while preserving were they are in the list of columns?

**Comment by YaLTeR** [comment]
I feel like some ways to manipulate workspaces are needed indeed, but I was thinking more of swapping or inserting a new one in between others. Haven't really had a need to merge workspaces yet.

**Reply by egrieco** [reply]
Found this comment looking through the discussions to see if there was a way to move a window up or down into a brand new workspace. This is already possible to do via the mouse in the overview, but I couldn't find a command to do the same.

**Reply by heyoeyo** [reply]
There's a way to do this through the IPC (see #2680, which is for moving 'up' when on the first workspace, but could be modified for moving up or down).

**Comment by wesbluemarine** [comment]
l

> whats your thoughts on adding functionallity to combine workspaces of windows, while preserving were they are in the list of columns?

Looks like something like dwm tag sytem.

**Reply by AlexD97** [reply]
Actually, I would like to see something like the dwm tag system in niri. However, I'm not sure, how this should behave in practice.

**Comment by AlexD97** [comment]
I would like to see a merge function where all windows from the workspace below are consumed into the columns above (and an inverse function that expels these windows again into their own workspace).

**Reply by YaLTeR** [reply]
So like you'd end up with columns of two windows? What if the count doesn't match?

**Reply by AlexD97** [reply]
> So like you'd end up with columns of two windows? 

Yeah, exactly.

> What if the count doesn't match?

I don't know, but it would probably be most consistent if only the upper or lower half of these columns are used.

**Comment by randoragon** [comment]
I think an action to append/prepend/insert the current workspace into another workspace would be cool. For example, you are using workspace 1 to work on a project. Then you move to workspace 2 for some extra work. Halfway through you realize that this is actually related to the same project, so you send the entire workspace to workspace 1, ending up with a concatenation of the two like this: [1:2].

I'm not sure how often this would be useful, but it seems like something that would be easy to implement. Of course, there would be no undo action, so if you merge by accident then you need to clean up manually :P
